systemdev/mms-infrastructure.rst

```diff
@@ -0,0 +1,82 @@
+MMS infrastructure components:
+==============================
+
+
+**oFono [1]** - responsible for providing the data context used to transfer mms data (image/music). It also propagates wap push notifications to upper layers.
+
+**nuntium [2]** - daemon that listens to wap push notifications and activates the mms data context on ofono on demand to send/receive mms's. It provides a local store, which will probably not be used by ubuntu-touch. The history service component will be used instead.
+
+**telepathy-ofono [3]**  - talks to nuntium through dbus and will be used to both relay messages from phone-app to nuntium and inject into the telepathy infrastructure mms's received by nuntium as multi-part messages [4]. This component also needs to mark messages as read and delete successfully received messages from nuntium.
+
+**history-service [5]**  - this component will watch the telepathy communication and store the messages received by telepathy-ofono or sent by phone-app.
+
+**telephony-service-approver [6]**  - this component is currently in charge of adding sms's to the messaging menu and to display incoming text notifications. It needs to be changed in order to add the received mms's to the messaging menu and display a notification that a new mms is available for download when auto-retrieval is disabled.
+
+**messaging-app  [7]** - It needs to correctly render mms's in the conversation view and also provide a way to attach media files to a message in order to send an mms.
+
+[1] - `hhttps://github.com/ubports/ofono <https://github.com/ubports/ofono>`_
+
+[2] - `https://github.com/ubports/nuntium <https://github.com/ubports/nuntium>`_
+
+[3] - `https://github.com/ubports/telepathy-ofono <https://github.com/ubports/telepathy-ofono>`_
+
+[4] - `https://telepathy.freedesktop.org/spec/Channel_Interface_Messages.html <http://telepathy.freedesktop.org/spec/Channel_Interface_Messages.html#Mapping:Message_Part>`_
+
+[5] - `https://github.com/ubports/history-service <https://github.com/ubports/history-service>`_
+
+[6] - `https://github.com/ubports/telephony-service <https://github.com/ubports/telephony-service>`_
+
+[7] - `https://github.com/ubports/messaging-app <https://github.com/ubports/messaging-app>`_
+
+
+
+General Description
+-------------------
+
+
+Incoming MMS:
+^^^^^^^^^^^^^
+
+
+* An mms WAP push notification is received by ofono and propagated to nuntium
+* nuntium enables the mms context in ofono, retrieves the content from MMSC and propagates the new MMS to upper layers.
```

Comment:
do you know what MMSC is?

Reply:
i've added a link to the MMS infrastructure on Wikipedia where all is explained here